File: vaadin-book-library/src/main/java/com/ufasoli/vaadin/library/model/Book.java
package com.ufasoli.vaadin.library.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.util.Date;

/**
 * Created with IntelliJ IDEA.
 * User: Ulises Fasoli
 * Date: 24.10.13
 * Time: 10:21
 */
@Document
public class Book implements Serializable {

    @Id
    private String bookId;
    private String bookTitle;
    private String bookCategory;
    private String userBorrowed;
    private Date borrowedOn;


    public Book() {
    }

    public Book(String bookId, String bookTitle, String bookCategory, String userBorrowed, Date borrowedOn) {
        this.bookId = bookId;
        this.bookTitle = bookTitle;
        this.bookCategory = bookCategory;
        this.userBorrowed = userBorrowed;
        this.borrowedOn = borrowedOn;
    }

    public Book(String bookId, String bookTitle, String bookCategory) {
        this.bookId = bookId;
        this.bookTitle = bookTitle;
        this.bookCategory = bookCategory;
    }

    public String getBookId() {
        return bookId;
    }

    public void setBookId(String bookId) {
        this.bookId = bookId;
    }

    public String getBookTitle() {
        return bookTitle;
    }

    public void setBookTitle(String bookTitle) {
        this.bookTitle = bookTitle;
    }

    public String getBookCategory() {
        return bookCategory;
    }

    public void setBookCategory(String bookCategory) {
        this.bookCategory = bookCategory;
    }

    public String getUserBorrowed() {
        return userBorrowed;
    }

    public void setUserBorrowed(String userBorrowed) {
        this.userBorrowed = userBorrowed;
    }

    public Date getBorrowedOn() {
        return borrowedOn;
    }

    public void setBorrowedOn(Date borrowedOn) {
        this.borrowedOn = borrowedOn;
    }
}

File: play-book-library/app/models/Book.scala
package models

import play.api.libs.json.Json
import com.novus.salat.global._

import play.api.Play.current
import java.util.Date
import com.novus.salat.annotations._
import com.novus.salat.dao._
import se.radley.plugin.salat._
import mongoContext._
import com.mongodb.casbah.commons.MongoDBObject
import org.bson.types.ObjectId

/**
 * Created with IntelliJ IDEA.
 * User: Ulises Fasoli
 * Date: 28.10.13
 * Time: 08:26
 *
 */
case class Book(@Key("_id") id: ObjectId = new ObjectId, bookId: String = "", bookTitle: String, bookCategory: String, userBorrowed: Option[String], borrowedOn: Option[Date]) {
  // implicit val bookFormat = Json.format[Book]
}


object Book extends ModelCompanion[Book, ObjectId] {
  def collection = mongoCollection("book")
  val dao = new SalatDAO[Book, ObjectId](collection =collection) {}


  def findAllFilterTitle(filter: String) = {
    println(MongoDBObject("bookTitle" ->
      MongoDBObject("$regex" -> s".*$filter.*")))

    dao.find(
      MongoDBObject("bookTitle" ->
        MongoDBObject("$regex" -> s".*$filter.*")
      ))
  }

  def findUserBorrowed(username:String) = dao.find(MongoDBObject("userBorrowed" -> username))


}


